cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflow.java

```diff
@@ -126,4 +127,26 @@ protected final void addSpark(String spark) {
     Predicate<WorkflowContext> predicate) {
     return configurer.condition(predicate);
   }
+
+  /**
+   * Adds a local dataset instance to the {@link Workflow}.
+   * See {@link co.cask.cdap.api.dataset.DatasetDefinition} for details.
+   *
+   * @param datasetName name of the dataset instance
+   * @param typeName name of the dataset type
+   * @param properties dataset instance properties
+   */
+  protected final void createLocalDataset(String datasetName, String typeName, DatasetProperties properties) {
```

Comment:
Why we only accept `typeName` as string, while in the normal `createDataset` call, we accept string name as well as `Class`?